### PR TITLE
Make Comdb2 server build without SSL/TLS support.

### PR DESCRIPTION
**Pull request by riverszhang89**
The pull request fixes server compilation when `-DWITH_SSL=0`. `WITH_SSL` is also made a CMake option to allow easy configuration.

Two symbols, `CLIENT_ssl_new_ctx` and `CLIENT_ssl_init`, are removed from `cdb2api.o`. Instead they're compiled into `ssl_support.o`.

### CODE REVIEW

**Comment by adizaimi** [comment]
This looks good to me.

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,13 @@ endif()
 #-g
 #endif()
 
+option(WITH_SSL "Turn OFF to compile without SSL" ON)
```
Comment (nirbhayc) [inline CMakeLists.txt]: The check below is redundant. OPTION() should take care of enabling/disabling the option.

Comment (riverszhang89) [reply CMakeLists.txt]: I don't think `option()` automatically adds a compile definition. The check is actually necessary.

Comment (nirbhayc) [reply CMakeLists.txt]: I stand corrected. Thank you.

**Comment by riverszhang89** [comment]
@adizaimi @nirbhayc Thank you for the review.

**Pull request merged**